File: scripts/gen_data.mjs
#!/usr/bin/env node
/* Regenera web/circuitos.data.js a partir de datos/circuitos.json (sin Python). */
import fs from 'node:fs'
import path from 'node:path'
import { fileURLToPath } from 'node:url'

const RAIZ = path.dirname(path.dirname(fileURLToPath(import.meta.url)))
const data = JSON.parse(fs.readFileSync(path.join(RAIZ, 'datos', 'circuitos.json'), 'utf8'))
const out =
  '/* Generado desde datos/circuitos.json — NO editar a mano. */\n' +
  'window.CIRCUITOS = ' + JSON.stringify(data, null, 2) + ';\n'
fs.writeFileSync(path.join(RAIZ, 'web', 'circuitos.data.js'), out)
console.log('OK -> web/circuitos.data.js')

File: scripts/gen_data.py
#!/usr/bin/env python3
"""Regenera web/circuitos.data.js a partir de datos/circuitos.json."""
import json, pathlib
root = pathlib.Path(__file__).resolve().parent.parent
d = json.loads((root/"datos/circuitos.json").read_text(encoding="utf-8"))
out = "/* Generado desde datos/circuitos.json — NO editar a mano. */\n"
out += "window.CIRCUITOS = " + json.dumps(d, ensure_ascii=False, indent=2) + ";\n"
(root/"web/circuitos.data.js").write_text(out, encoding="utf-8")
print("OK")
